### ISSUE TITLE
pkcs#8 output needed for resource tls_private_key

### ISSUE THREAD

**Issue by stvdilln** [hashicorp/terraform-provider-tls#91]
I'm trying to import a key created with tls_private_key (with the private_key_pem attribute) into the azurerm_key_vault_certificate resource.  azurerm_key_vault_certificate rejects the private key created with tks_private_key as it in in pkcs#1 format and not the pkcs#8 format that Azure Key Vault wants.

tls_private_key.private_key_pem attrtibute outputs a PEM with '-----BEGIN RSA PRIVATE KEY-----' header which is PKCS#1, azurerm_key_vault_certificate expects a cert package with a private key that has a pkcs#8 header '-----BEGIN PRIVATE KEY-----'

If I convert the output of tls_private_key.private_key_pem with openssl (openssl pkcs8 -topk8 -inform PEM -outform PEM -nocrypt -in ca-key.pem -out test-key.pem) then I can import the certificate fine.

The snippet of code that is of interest:
```
resource tls_private_key ca_key {
   algorithm = "RSA"
   rsa_bits  = 4096
}
```
This relates to [#3017 on azurerm provider](https://github.com/terraform-providers/terraform-provider-azurerm/issues/3017).  In that issue it was closed due to Azure Issue.  Azure has fixed their backend, so current state is when I try to import the private 
key into azurerm_key_vault_certificate I get the following error:
The specified PEM X.509 certificate content is in an unexpected format. Please check if certificate is in valid PEM format.

### Terraform Version
Terraform version 0.13.5
TLS provider version 3.0

### Affected Resource(s)
- tls_private_key
- azurerm_key_vault_certificate

### Terraform Configuration Files
```hcl
variable private_key_size {
    default = 4096
}
variable private_key_algorithim {
    default = "RSA"
}
resource tls_private_key ca_key {
   algorithm = var.private_key_algorithim
   rsa_bits  = var.private_key_size
}


resource tls_self_signed_cert ca_cert {
   private_key_pem = tls_private_key.ca_key.private_key_pem
   key_algorithm = "RSA"
   subject {
     common_name         = var.common_name
     organization        = var.issuer_organization.organization
     organizational_unit = var.issuer_organization.organizational_unit
     street_address      = var.issuer_organization.street_address
     locality            = var.issuer_organization.locality
     province            = var.issuer_organization.province
     country             = var.issuer_organization.country
     postal_code         = var.issuer_organization.postal_code

   }
   # 175200 = 20 years
   validity_period_hours = 175200
   allowed_uses = [
     "cert_signing",
     "crl_signing"
   ]
   is_ca_certificate = true 

}
resource local_file private_key {
    sensitive_content = tls_private_key.ca_key.private_key_pem
    filename = "${var.output_cert_path}/ca-key.pem"
    file_permission = "0600"
}
resource local_file ca_file {
    sensitive_content = tls_self_signed_cert.ca_cert.cert_pem
    filename = "${var.output_cert_path}/ca-cert.pem"
    file_permission = "0600"
}
resource local_file ca_pem_bundle {
    sensitive_content = "${tls_private_key.ca_key.private_key_pem}${tls_self_signed_cert.ca_cert.cert_pem}"
    filename = "${var.output_cert_path}/ca-bundle.pem"
    file_permission = "0600"
}

resource azurerm_key_vault_certificate ca_cert {
  name          = var.service_settings.cert_name
  key_vault_id  = var.service_settings.key_vault_resource_id

  certificate {
    contents = "${tls_private_key.ca_key.private_key_pem}${tls_self_signed_cert.ca_cert.cert_pem}"
    #contents = file("./secrets/test.pem")
    password = ""
  }
  certificate_policy {
    key_properties {
      exportable = "true"
      key_size   = var.private_key_size 
      key_type   = var.private_key_algorithim
      reuse_key  = "true"
    }
    issuer_parameters {
      name = "Self"
    }
    secret_properties {
      content_type = "application/x-pem-file"
    }
  }



}
```

### Debug Output
https://gist.github.com/stvdilln/b1bf0ae3b68b64b5ad7ddf3ec1e44b9b

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
I should be able to take key from tls_self_signed_cert and import into azurerm_key_vault_certificate.

### Actual Behavior
Error about cert package:
```
 The specified PEM X.509 certificate content is in an unexpected format. Please check if certificate is in valid PEM format.
```
### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### References
AzureRM provider issue 3017.[
Microsoft Docs](https://docs.microsoft.com/en-us/azure/key-vault/certificates/certificate-scenarios#formats-of-import-we-support)
Microsoft docs snippet:
```
Azure Key Vault supports .pem and .pfx certificate files for importing Certificates into Key vault. We support the following type of Import for PEM file format. A single PEM encoded certificate along with a PKCS#8 encoded, unencrypted key which has the following
```

**Comment by BojanZelic** [comment]
:+1: 

I'm running into the same issue when trying to generate cloudflare argo certificates. 
The Private key is required to be in PKCS#8 format. 

The tls_cert_request module also does not seem to support a PKCS#8 key format and errors out with the following:
```
resource "tls_cert_request" "example" {
  key_algorithm   = "RSA"
  private_key_pem = file("./private_pkcs8.key")
  subject {
    common_name  = "example"
    organization = "example"
  }
}
```

```
Error: x509: failed to parse private key (use ParsePKCS8PrivateKey instead for this key format)
```

**Comment by detro** [comment]
Fixed since 3.2.1 (see #182 and related issues)
